ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -441,6 +490,42 @@ def record_usage(self, span: Span, usage: Dict[str, Any]) -> None:
         if total_cost:
             self.metric(span, "incr", "tokens.total_cost", total_cost)
 
+    def check_token_usage_chat_or_llm_result(self, result):
+        """Checks for token usage on the top-level ChatResult or LLMResult object"""
+        llm_output = getattr(result, "llm_output", None) or {}  # in case it is explicitly set to None
+        token_usage = (
+            llm_output.get("token_usage", {})
+            or llm_output.get("usage_metadata", {})
+            or llm_output.get("usage", {})
+            or {}  # in case one is explicitly set to None
```

Comment:
Are these mutually exclusive? i.e. llm_output will only have one of these filled while the others are None or don't exist? If so, thoughts on chaining the get default value instead of the OR operation?

```python
llm_output.get("token_usage", llm_output.get("usage_metadata", llm_output.get("usage", {})))
```

Reply:
Ah good point, it should only ever have one, the name just depends on the partner library. Let me change this logic (can maybe do this in a couple other spots), and see that it still looks good.

Reply:
i changed the logic a bit, but still added an extra `dict` check to be safe, since we're technically getting these from partner libraries, seemingly without standardization, I just wanna be extra safe we don't crash. 

Also, for some reason, sometimes these fields are explicitly set to `None`, so they don't fall back on the default value of `dict.get(..., default)`, so I added those extra `is not None` checks to be safe. 

tl;dr check everything to be safe with langchain partner libraries 😅